src/pages/index.js

```diff
@@ -9,49 +9,90 @@ import Card from "../components/card.js";
 import FormValidator from "../components/FormValidator.js";
 import PopupWithForm from "../components/PopupWithForm.js";
 import PopupWithImage from "../components/PopupWithImage.js";
+import PopupConfirm from "../components/PopupConfirm.js";
 import UserInfo from "../components/UserInfo.js";
+import Api from "../components/Api.js";
 
 /* -------------------------------- #VARIABLES# ------------------------------- */
 
 /* ---------------------------------- UTILS --------------------------------- */
 import { validationConfig } from "../utils/utils.js";
 
-/* ---------------------------------- DATA ---------------------------------- */
-import { cardData } from "../utils/Constants.js";
-
 /* ------------------------------ EDIT PROFILE ------------------------------ */
 import {
   profileEditButton,
   profileEditModal,
-  closeButton,
   profileTitle,
   profileDescription,
   profileTitleInput,
   profileDescriptionInput,
+  profilePicture,
 } from "../utils/Constants.js";
 
 /* -------------------------------- ADD CARD -------------------------------- */
+import { profileAddModal, profileAddButton } from "../utils/Constants.js";
+
+/* ------------------------- CHANGE PROFILE PICTURE ------------------------- */
 import {
-  modalAddCloseButton,
-  profileAddModal,
-  profileAddButton,
-  profileAddForm,
-  cardTitleInput,
-  cardUrlInput,
+  profilePictureModal,
+  profileEditPictureButton,
 } from "../utils/Constants.js";
 
-/* ------------------------------ PREVIEW CARD ------------------------------ */
-import { modalImageCloseButton } from "../utils/Constants.js";
-
 /* ----------------------------------- GRID/CARDS ---------------------------------- */
-import { galleryListEl } from "../utils/Constants.js";
+import { cardLikeIcon } from "../utils/Constants.js";
+
+/* ---------------------------------- Modal --------------------------------- */
+import { submitButton } from "../utils/Constants.js";
 
 /* -------------------------------------------------------------------------- */
 /*                                  FUNCTIONS                                 */
 /* -------------------------------------------------------------------------- */
+
+const apiUrl = "https://around-api.en.tripleten-services.com";
+
+const newApi = new Api(apiUrl);
+
+const confirmPopup = new PopupConfirm("#confirm-modal");
+
+confirmPopup.setEventListeners();
+
 function createCard(items) {
-  const newCard = new Card("#card-template", items, handleCardClick);
-  const cardElement = newCard.getTemplate();
+  const card = new Card(
+    "#card-template",
+    items,
+    handleCardClick,
+    function (card) {
+      confirmPopup.openPopup();
+      confirmPopup.handleConfirm(function () {
+        const cardId = items._id;
+        newApi.deleteApiCard(cardId).then(card);
+        card.deleteCard();
```

Comment:
You need to **delete** cards only in block `then` after a **successful** response from the server so that if there is an error the card would stay in the `DOM`.